core/src/G3Frame.cxx

```diff
@@ -149,7 +150,8 @@ static std::string FrameObjectClassName(G3FrameObjectConstPtr obj)
 		}
 	}
 
-	return cxx_demangle(typeid(*obj).name());
+	auto &r = *obj;
+	return cxx_demangle(typeid(r).name());
```

Comment:
Why does this help? I don't see an obvious reason to warn about the original code...

Reply:
The warning was:

```
/Users/arahlin/code/spt/spt3g_software_public/core/src/G3Frame.cxx:152:29: warning: expression with side effects will be evaluated despite being used as an operand to 'typeid' [-Wpotentially-evaluated-expression]
        return cxx_demangle(typeid(*obj).name());
                                   ^
```

Reply:
I've pushed a different solution that templates `cxx_demangle` instead, based on one of the comments in [this](https://stackoverflow.com/questions/46494928/clang-warning-on-expression-side-effects) post.

Reply:
Weird, OK. Thanks!